app/controllers/stages_controller.rb

```diff
@@ -102,4 +99,12 @@ def stage_params
   def find_stage
     @stage = current_project.stages.find_by_param!(params[:id])
   end
+
+  def template_param
+    @template_param ||= ActiveRecord::Type::Boolean.new.type_cast_from_database(stage_params.delete("template"))
+  end
+
+  def update_template
+    template_param ? @stage.project.template_stage!(@stage) : @stage.project.reset_template_stage!
+  end
```

Comment:
this still looks wrong to me:
 - removing the template param -> remove the attribute -> simple assignment
 - not changing then template param -> no change ... bug: atm it unsets the template
 - adding template param -> set the attribute -> can be a assignment + after-save remove it from all previous stage